Sources/Core/StaticString.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+@frozen
+public struct StaticString {
+  /// `StaticString` stores either a pointer to null-terminated string data, or a single unicode scalar if possible.
+  @usableFromInline
+  internal var _startPtrOrData: Builtin.Word
+
+  @usableFromInline
+  internal var _utf8CodeUnitCount: Builtin.Word
+
+  /// Flags indicating how `StaticString` storage is used.
```

Comment:
Having additional documentation of reserved and in use fields would be desirable.